src/clean_acquisitions_data.py

```diff
@@ -0,0 +1,79 @@
+import logging
+import os
+import re
+import argparse
+
+import yaml
+import pandas as pd
+import boto3
+
+from helpers import gen_helpers as gen_h
+
+logging.basicConfig(format="%(asctime)s - %(name)s - %(levelname)s - %(message)s")
+logger = logging.getLogger(__name__)
+
+def get_acquisition_count(df, all_companies_file):
+    """Aggregates acquisition by acquirer and creates new dataframe for
+    acquisition count"""
+    # Create list of acquired companies in original company dataset
+    company_df = gen_h.read_data(all_companies_file)
+    all_companies_list = list(company_df['permalink'].unique())
+    acquired_companies_list = list(df['company_permalink'].unique())
+    acquirer_companies_list = list(df['acquirer_permalink'].unique())
+
+    acquired_companies_list = [permalink for permalink in acquired_companies_list
+    if permalink in all_companies_list]
+    acquirer_companies_list = [permalink for permalink in acquirer_companies_list
+    if permalink in all_companies_list]
+
+    # Create dictionary of number of acquisitions by acquirers that are in original company dataset
+    acquirer_dict = dict(df['acquirer_permalink'].value_counts())
+    acquirer_dict = {acquirer: acquirer_dict[acquirer] for acquirer in acquirer_dict
+    if acquirer in acquirer_companies_list}
+
+    # Create dataframe of number of acquisitions to merge into other dataframes
+    acquirers_df = pd.DataFrame.from_dict(acquirer_dict, orient='index').reset_index()
+    acquirers_df.columns = ['permalink', 'no_acquisitions']
+    return acquirers_df
+
+def run_clean_acquisitions(args):
+    """Loads config and cleans the companies dataset. Uses cleaned data to
+    generate new features.
+    Args:
+        args: From argparse, should contain args.config and optionally, args.save
+            args.config (str): Path to yaml file with load_data as a top level
+            key containing relevant configurations
+            args.save (str): If given, resulting dataframe will be saved to this
+            location.
+    Returns: None
+    """
+    with open(args.config, "r") as f:
+        config = yaml.load(f)
+
+    df = gen_h.read_data(args.input_file)
+    df = get_acquisition_count(df, **config['clean_acquisitions_data']
+    ['get_acquisition_count'])
+
+    df.to_csv(args.save)
```

Comment:
do you mean just to check if the file path exists?

Reply:
`to_csv` requires a valid path right? like if they gave `~/some/dir/that/doesnt/exist/file.csv` it would fail right? or you could strip the specified path down to just the file name so that user is not able to write outside of current working dir - yanno? Just in general good to validate/clean anything that user inputs